docs/fn-2.0/README.md

```diff
@@ -0,0 +1,416 @@
+# Fn 2.0
+
+## NOTE: This is a proposal for updating Fn with the following goals:
+
+* Split HTTP routes from functions
+* Support CloudEvent messages as the native format throughout
+* Simplify everything
+* Modularize components
+* Namespacing
+
+## Removals
+
+* Default, HTTP and JSON formats will be removed
+* Cold functions will be removed
```

Comment:
How are functions/containers not built with an FDK supported, i.e. ones that accept STDIN only? I've read or been part of discussions around creating `hot wrappers` that get injected or wrapped around a "cold" container to turn it into a "hot" one - think @shaunsmith had something almost working - but no final solution or proposal yet.

Reply:
At least initially the plan here was: 

![giphy](https://user-images.githubusercontent.com/2935111/39328886-9cd03e98-4951-11e8-9a2a-dcbcc8256ba2.gif)

we had some discussion around letting functions shoot themselves via extensions to get 'cold' behavior, but that would not itself solve the 'take any function that does stdin / stdout and run it like normal for 1 invocation', it would only solve the 'cold' thing.

seemingly, every other FaaS product has its own format a user must use for input/output and we would just be joining the club. Seems like we're mostly re-opening https://github.com/fnproject/fn/issues/785 discussion here, which was contentious. it would be really nice to only have 1 format, there's been quite a large amount of pain supporting multiple formats and heat levels, over time between the backend as well as for FDKs, CLI, and simply to users.

Reply:
Yeah there is a lot of conflicting users stories around Fn going around right now.

Reply:
The conversation I had with @shaunsmith, we were talking about whether we can have a small binary that gets injected into a container (or the original container gets wrapped) and becomes the new entry point, takes the json/cloudevent data off the wire and then marshalls it to stdin, launching the original function entry point as a process in-container, effectively achieving cold-like behaviour from the function's perspective.  The stdin/stdout any-container any-language pitch always plays strongly at the meetups.

Reply:
That maybe true but the container still has to be written to use our signatures no matter what wrapping is done so it still does not provide and real value IMO. Any container may be a function but not not all containers are functions. Where as all functions are containers. Squares and Rectangles.

Reply:
@rikgibson that seems like a good plan, openfaas does similar to that. that pushes that logic to have about as small of a surface area as possible, the functions handling it individually.